Basic-Car-Maintenance/Shared/Odometer/Views/OdometerView.swift

```diff
@@ -14,6 +14,8 @@ struct OdometerView: View {
     
     @State private var viewModel: OdometerViewModel
     @State private var selectedTimeRange: TimeRange = .all
+    // start of my variables
+    @State private var selectedVehicleID: String?
```

Comment:
no need for this comment also a variable for `selectedVehicleID` was created in the `OdometerViewModel` so it should be used, rather than making a new variable in the view, which would lead to two different sources of truth

And this is a string, which makes sense for specifically keeping track of the vehicleID, but then later setting something to `= "nil"` doesn't feel right cause that does have a value and feels like a code smell